src/Microsoft.SqlTools.ServiceLayer/EditData/EditDataService.cs

```diff
@@ -0,0 +1,328 @@
+            serviceHost.SetRequestHandler(EditUpdateCellRequest.Type, HandleUpdateCellRequest);
+
+            // Register handler for shutdown event
+            serviceHost.RegisterShutdownTask((shutdownParams, requestContext) =>
+            {
+                Dispose();
+                return Task.FromResult(0);
+            });
+        }
+
+        #region Request Handlers
+
+        internal async Task HandleCreateRowRequest(EditCreateRowParams createParams,
+            RequestContext<EditCreateRowResult> requestContext)
+        {
+            try
+            {
+                Session session = GetActiveSessionOrThrow(createParams.OwnerUri);
+
+                // Create the row and get send the ID of the row back
+                long newRowId = session.CreateRow();
+                EditCreateRowResult createResult = new EditCreateRowResult
+                {
+                    NewRowId = newRowId
+                };
+                await requestContext.SendResult(createResult);
+            }
+            catch (Exception e)
+            {
+                await requestContext.SendError(e.Message);
+            }
+        }
+
+        internal async Task HandleDeleteRowRequest(EditDeleteRowParams deleteParams,
+            RequestContext<EditDeleteRowResult> requestContext)
+        {
+            try
+            {
+                Session session = GetActiveSessionOrThrow(deleteParams.OwnerUri);
+
+                // Add the delete row to the edit cache
+                session.DeleteRow(deleteParams.RowId);
+                await requestContext.SendResult(new EditDeleteRowResult());
+            }
+            catch (Exception e)
+            {
+                // Send back the error
+                await requestContext.SendError(e.Message);
+            }
+        }
+
+        internal async Task HandleDisposeRequest(EditDisposeParams disposeParams,
+            RequestContext<EditDisposeResult> requestContext)
+        {
+            try
+            {
+                // Sanity check the owner URI
+                Validate.IsNotNullOrWhitespaceString(nameof(disposeParams.OwnerUri), disposeParams.OwnerUri);
+
+                // Attempt to remove the session
+                Session session;
+                if (!ActiveSessions.TryRemove(disposeParams.OwnerUri, out session))
+                {
+                    await requestContext.SendError(SR.EditDataSessionNotFound);
+                    return;
+                }
+
+                // Everything was successful, return success
+                await requestContext.SendResult(new EditDisposeResult());
+            }
+            catch (Exception e)
+            {
+                await requestContext.SendError(e.Message);
+            }
+        }
+
+        internal async Task HandleInitializeRequest(EditInitializeParams initParams,
+            RequestContext<EditInitializeResult> requestContext)
+        {
+            try
```

Comment:
As mentioned above, I've found a handy pattern is to have 1 "HandleRequest" method that does the error handling & returning of messages over the channel. I'd suggest using it here instead of the try/catch. See TSqlFormatterService.HandleRequest<T>()